crates/cow-amm/src/implementations/safe_based/cow_amm_constant_product_factory.rs

```diff
@@ -0,0 +1,58 @@
+use {
+    crate::{implementations::safe_based::cow_amm::CowAmm, indexer::CowAmmRegistry},
+    ethcontract::{common::DeploymentInformation, dyns::DynWeb3},
+    shared::impl_event_retrieving,
+    std::sync::Arc,
+};
+
+impl_event_retrieving! {
+    pub Contract for contracts::cow_amm_constant_product_factory
+}
+
+pub struct CowAmmConstantProductFactoryHandler {
+    contract: contracts::CowAmmConstantProductFactory,
+}
+
+impl CowAmmConstantProductFactoryHandler {
+    pub fn from_contract(contract: contracts::CowAmmConstantProductFactory) -> Arc<Self> {
+        Arc::new(Self { contract })
+    }
+
+    pub async fn deployed(web3: &DynWeb3) -> Arc<Self> {
+        Arc::new(Self {
+            contract: contracts::CowAmmConstantProductFactory::deployed(web3)
+                .await
+                .expect("Failed to find deployed CowAmmConstantProductFactory"),
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl crate::ContractHandler<contracts::cow_amm_constant_product_factory::Event>
```

Comment:
Let's import `contracts::cow_amm_constant_product_factory::Event` so we don't have such long names.